compiler/rustc_const_eval/src/interpret/cast.rs

```diff
@@ -356,7 +356,7 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             }
 
             _ => {
-                span_bug!(self.cur_span(), "invalid unsizing {:?} -> {:?}", src.layout.ty, cast_ty)
+                throw_inval!(Unsize(src.layout.ty, cast_ty))
```

Comment:
Even more const-prop induced technical debt in the interpreter. :(

How can this one even happen...?

Reply:
We can have a generic context
```rust
fn foo<T: ?Sized, V: std::marker::Unsize<T>>(value: &V) -> &T { value }
```

Reply:
Please add a comment explaining that (and that this is only needed for ConstProp).

Why does that not affect the similar `bug!` in unsize_into?